oop2-encapsulate-inherit.js

```diff
@@ -5,27 +5,269 @@
+        }
+
+        downsizing(){
+            console.log(`${this.#compName} is closing the ${this.#location} branch.`);
+            this.#inBusiness = false;
+        }
+
+        businessStatus(){
+            return this.#inBusiness;
+        }
+    }
+
+    class Employee extends Site {
+        #employeeName;
+        #jobTitle;
+        #salary;
+        #gender;
+        static demerits = [];
+
+        constructor(compName, location, employeeName, jobTitle, salary, gender){
+            super(compName, location);
+            this.#employeeName = employeeName;
+            this.#jobTitle = jobTitle;
+            this.#salary = salary;
+            this.#gender = gender;
+        }
+
+        approveRaise(raiseAmt){
+            this.#salary += raiseAmt;
+        }
+
+        checkSalary(){
+            return this.#salary;
+        }
+
+        writeUp(issue){
+            Employee.demerits.push(issue);
+            console.log(`${this.#employeeName} has been given a demerit for ${issue}.`);
+        }
+
+        checkDemerits(){
+            return Employee.demerits;
+        }
+    }
+
+    const allegis = new Umbrella ('Allegis');
+
+    const tekSystems = new Company ('TEKsystems', 1500, 'Code all day!', 1999);
+    const aerotek = new Company ('Aerotek', 2400, 'We code in space!', 1983);
+    const tekGlobal = new Company ('TEK Global', 35000, 'We do it all!', 1987);
+
+    const tekIrving = new Site ('TEKsystems', 'Irving, TX');
+    const tekJuno = new Site ('TEKsystems', 'Juno, AK');
+    const tekSherman = new Site ('TEKsystems', 'Sherman, TX');
+
+    const aeroPlano = new Site ('Aerotek', 'Plano, TX');
+    const aeroSanFran = new Site ('Aerotek', 'San Fransisco, CA');
+    const aeroMiami = new Site ('Aerotek', 'Miami, FL');
+
+    const globalHouston = new Site ('TEK Global', 'Houston, TX');
+    const glonbalLondon = new Site ('TEK Global', 'London, England');
+    const globalIceland = new Site ('TEK Global', 'Iceland');
+
+    const managerMichael = new Employee ('TEKsystems', 'Irving, TX', 'Michael', 'Regional Manager', 120000, 'Male');
+    const asstManagerDwight = new Employee ('TEKsystems', 'Irving, TX', 'Dwight', 'Assistant to the Regional Manager', 100000, 'Male');
+    const custServiceKelly = new Employee ('TEKsystems', 'Sherman, TX', 'Kelly', 'Customer Service', 80000, 'Female');
+    const hrToby = new Employee ('Aerotek', 'Plano, TX', 'Toby', 'Human Resources', 90000, 'Male');
+    const salesJim = new Employee ('Aerotek', 'Miami, FL', 'Jim', 'Sales', 85000, 'Male');
+    const accAngela = new Employee ('Aerotek', 'San Fransisco, CA', 'Angela', 'Accountant', 95000, 'Female');
+    const lawyerKyle = new Employee ('TEKGlobal', 'Houston, TX', 'Kyle', 'Lawyer', 110000, 'Male');
+    const daWalter = new Employee ('TEK Global', 'London, England', 'Walter', 'Data Analysis', 105000, 'Male');
+    const tempRyan = new Employee ('TEK Global', 'Iceland', 'Ryan', 'Temp Employee', 55000, 'Male');
+
+
```

Comment:
use some of these objects to console log some things and prove to yourself it works